cli/command.go

```diff
@@ -274,19 +274,49 @@ func (c *BaseCommand) Hidden() bool {
 
 // Prompt prompts the user for a value. If stdin is a tty, it prompts. Otherwise
 // it reads from the reader.
-func (c *BaseCommand) Prompt(msg string) (string, error) {
-	scanner := bufio.NewScanner(io.LimitReader(c.Stdin(), 64*1_000))
+func (c *BaseCommand) Prompt(ctx context.Context, msg string, args ...any) (string, error) {
+	resultCh := make(chan string, 1)
+	errCh := make(chan error, 1)
+
+	// Prompts for user input and returns the value or an error if there is one.
+	// If the context is canceled, this function leaves the c.Stdin in a bad state.
+	// This is currently acceptable as the cli process should ideallly be terminated
+	// after the context is canceled.
+	go func() {
+		defer close(resultCh)
+		defer close(errCh)
+
+		scanner := bufio.NewScanner(io.LimitReader(c.Stdin(), 64*1_000))
+
+		if c.Stdin() == os.Stdin && isatty.IsTerminal(os.Stdin.Fd()) {
+			fmt.Fprint(c.Stdout(), msg, args)
+		}
 
-	if c.Stdin() == os.Stdin && isatty.IsTerminal(os.Stdin.Fd()) {
-		fmt.Fprint(c.Stdout(), msg)
-	}
+		scanner.Scan()
 
-	scanner.Scan()
+		if err := scanner.Err(); err != nil {
```

Comment:
I think this whole function can be simplified:

```
func (...) Prompt(...) string {
  if c.Stdin() == os.Stdin && isatty.IsTerminal(os.Stdin.Fd()) {
    fmt.Fprint(c.Stdout(), msg, args)
  }
  scanner := bufio.NewScanner(io.LimitReader(c.Stdin(), 64*1_000)
  finished := make(chan struct{})
  go func() {
    scanner.Scan()
    close(finished)
  }()
  
  select {
    case <-ctx.Done():
      return "", fmt.Errorf("failed to prompt: %w", ctx.Err())
    case <-finished:
  }
  
  if err := scanner.Err(); err != nil {
    return fmt.Errorf("failed to read stdin: %w", err)
  }
  return scanner.Text(), nil
}
```

Reply:
I'd probably do:

```
  go func() {
    defer close(finished)
    scanner.Scan()
  }()
```

but yea I think you're right

Reply:
Thanks for the input!